app/Http/Middleware/Role.php

```diff
@@ -0,0 +1,25 @@
+<?php
+
+namespace App\Http\Middleware;
+
+use Closure;
+
+class Role
+{
+    /**
+     * Handle an incoming request.
+     *
+     * @param  \Closure(\Illuminate\Http\Request): (\Symfony\Component\HttpFoundation\Response)  $next
+     */
+
+    public function handle($request, Closure $next, ...$roles)
+    {
+        foreach ($roles as $role) {
```

Comment:
- request user ở đây là auth e nhỉ, nếu vậy e dùng auth cho rõ code
- ở route e nên gom chung lại, trỏ tời cùng 1 middleware này, không cần truyền $role, trong đây mình sẽ tiến hành phân cấp để check permission, vd:
 + mình sẽ group các route lại, $adminAllowRoute = ['user.', 'post.'], $editorAllowRoute = ['post.'], có các block sau:
 if Route::currentRouteName() in adminAllowRoute && auth() == admin...
 if Route::currentRouteName() in editorAllowRoute && auth()  == editor ...
các block trên gom lại các private function, handle sẽ gọi ra để check, pass thì next fail thì 403

Reply:
Dạ em đã sửa theo fb này ạ https://github.com/anhlp1212/blog_laravel/pull/12/commits/0a2d99689ba6ad9d896b8b2753c3ab72b85276b4
Nhưng mà em chưa biết là đúng ý chưa nên nhờ anh nhận xét lại giúp em ạ.